### PR TITLE
Remove datagen JSON config

### PR DESCRIPTION
**Pull request by robertbastian**
Given that we're not sure about the `config.json` API, we can just use the Rust API for baked data. Having a single script also ensures that all baked data uses the same source data and locales, which is subject to human error at the moment.

#3960

### CODE REVIEW

**Comment by sffc** [comment]
OK

This PR can stand on its own merits outside of the config file discussion, and it doesn't close the door to revisiting the config file in the future

File: components/datetime/tests/data/gen.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+cargo run -p icu_datagen -- \
+--keys "datetime/gregory/datelengths@1" "datetime/gregory/datesymbols@1" "datetime/timelengths@1" "datetime/timesymbols@1" "decimal/symbols@1" "time_zone/formats@1" "time_zone/specific_short@1" \
```
Comment (sffc) [inline components/datetime/tests/data/gen.sh:3]: Question: Was this code being run in CI before, and it is being run in CI now?

File: experimental/transliterate/tests/data/gen.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
```
Comment (sffc) [inline experimental/transliterate/tests/data/gen.sh:1]: Also here

**Review by sffc — APPROVED** [review]

File: components/datetime/tests/data/gen.sh (hunk 1 above)
Comment (robertbastian) [reply components/datetime/tests/data/gen.sh:3]: it was not

File: experimental/transliterate/tests/data/gen.sh (hunk 2 above)
Comment (robertbastian) [reply experimental/transliterate/tests/data/gen.sh:1]: it was not

File: components/datetime/tests/data/gen.sh (hunk 1 above)
Comment (robertbastian) [reply components/datetime/tests/data/gen.sh:3]: and is not

**Pull request merged**